source/common/secret/secret_manager_impl.h

```diff
@@ -77,7 +81,8 @@ class SecretManagerImpl : public SecretManager {
     std::shared_ptr<SecretType>
     findOrCreate(const envoy::config::core::v3::ConfigSource& sds_config_source,
                  const std::string& config_name,
-                 Server::Configuration::TransportSocketFactoryContext& secret_provider_context) {
+                 Server::Configuration::TransportSocketFactoryContext& secret_provider_context,
```

Comment:
In a perfect world we would remove the init manager from the passed in context if it's not actually valid (so instead make some new type of context for the secret manager), but I would defer to @kyessenov and @alyssawilk who have more recent experience with all of the contexts and the right thing to do here. Otherwise this generally looks like the right track. Thanks!

Reply:
I don't have context on this particular change but
0) thank you for catching this and fixing this!
1) I have encountered the fact that the server context has an init manger being dangerous (I wonder if it's worth the churn to rename to something like startUpOnlyInitManager() but it might not have added clarity here)
2) definitely worth adding a reload test which would not have worked before (I think you can probably combine one of the lds integration tests with the oath tests?) and comments explaining why this is important

Reply:
Yes, I agree this is a good catch, and thanks for finding the root cause. I think this bug came to be because the original usage was only in the context of the upstream clusters and the downstream listeners, where the transport socket was the expected thing that needs to be initialized. So technically, the code isn't wrong and runtime detection wouldn't help. But I'd also recommend not sub-typing contexts (e.g. avoid passing any other context) to prevent lifetime issues.